debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/sink/StartedMongoDbSinkTask.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.sink;
+
+import static io.debezium.connector.mongodb.sink.MongoDbSinkConnectorTask.LOGGER;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.OptionalLong;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.bson.BsonDocument;
+
+import com.mongodb.MongoBulkWriteException;
+import com.mongodb.MongoNamespace;
+import com.mongodb.bulk.BulkWriteResult;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.WriteModel;
+
+import io.debezium.dlq.ErrorReporter;
+
+final class StartedMongoDbSinkTask implements AutoCloseable {
+
+    private final MongoDbSinkConnectorConfig sinkConfig;
+    private final MongoClient mongoClient;
+    private final ErrorReporter errorReporter;
+
+    StartedMongoDbSinkTask(
+                           final MongoDbSinkConnectorConfig sinkConfig,
+                           final MongoClient mongoClient,
+                           final ErrorReporter errorReporter) {
+        this.sinkConfig = sinkConfig;
+        this.mongoClient = mongoClient;
+        this.errorReporter = errorReporter;
+    }
+
+    @SuppressWarnings("try")
+    @Override
+    public void close() {
+        try (MongoClient autoCloseable = mongoClient) {
+        }
+    }
+
+    void put(final Collection<SinkRecord> records) {
+        try {
+            trackLatestRecordTimestampOffset(records);
+            if (records.isEmpty()) {
+                LOGGER.debug("No sink records to process for current poll operation");
+            }
+            else {
+                List<List<MongoProcessedSinkRecordData>> batches = MongoSinkRecordProcessor.orderedGroupByTopicAndNamespace(
+                        records, sinkConfig, errorReporter);
+                for (List<MongoProcessedSinkRecordData> batch : batches) {
+                    bulkWriteBatch(batch);
+                }
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Use `DebeziumException` where possible instead of `RuntimeException`

Reply:
done